client/src/use/useLineChart.ts

```diff
@@ -42,14 +42,15 @@ export default function useLineChart({
      * For each begin time, push a 1, for each end time, push a -1.
      * Then iterate each histogram and generate its accumulation at each point.
      */
-    enabledTracks.value.forEach((track) => {
+    enabledTracks.value.forEach((filtered) => {
+      const { track } = filtered;
       const totalArr = histograms.get('total') as number[];
       const ibegin = track.begin;
       const iend = track.end > track.begin ? track.end : track.begin + 1;
       [totalArr[ibegin], totalArr[iend]] = updateHistogram(ibegin, iend, totalArr);
-      const { confidencePairs } = track;
-      if (confidencePairs.length) {
-        const trackType = confidencePairs[0][0];
+      const type = track.getType(filtered.context.confidencePairIndex);
+      if (type && type.length) {
+        const trackType = type ? type[0] : '';
```

Comment:
```suggestion
        const trackType = type[0];
```

also too defensive.  This should throw an out-of-bounds if it's wrong, not fail silently.  Similar to above note about not having defaults for invalid states.